packages/remix-react/transition.ts

```diff
@@ -1366,11 +1366,6 @@ function filterMatchesToLoad(
     return true;
   };
 
-  let isInRootCatchBoundary = state.matches.length === 1;
-  if (isInRootCatchBoundary) {
-    return matches.filter(match => !!match.route.loader);
-  }
-
```

Comment:
looks like it was added in #264

Reply:
@jacob-ebey any idea what was for?

Reply:
yeah looks like we're still manually creating matches for that 
https://github.com/remix-run/remix/blob/960a560e79abe71e5519eb5111b8cc013c178f0a/packages/remix-react/transition.ts#L431-L437

https://github.com/remix-run/remix/blob/960a560e79abe71e5519eb5111b8cc013c178f0a/packages/remix-react/transition.ts#L502-L508

Reply:
alright, I've updated the `isInRootCatchBoundary` check instead of flat out removing it https://github.com/remix-run/remix/pull/1874/commits/03293570a7ab436da6eee4e84d0bacc60a9a12d2#diff-6dda36fa0981ac6fdbdb64f280958e54decf21c27d91a610e3359f02d06bc10fR1528-R1529

looking at a test now, but these cases seem covered by our current CatchBoundary tests